pkgs/tools/text/sorted-grep/default.nix

```
@@ -0,0 +1,44 @@
+{ stdenv, fetchurl, lib }:
+
+stdenv.mkDerivation rec {
+  pname = "sorted-grep";
+  version = "1.0";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/sgrep/sgrep-${version}.tgz";
+    hash = "sha256-3F7cXrZnB38YwE1sHYm/CIGKmG+1c0QU+Pk3Y53a0T4=";
+  };
+
+  postPatch = ''
+    # Its Makefile is missing compiler flags and an install step
+    rm -f Makefile
+  '';
+
+  buildPhase = ''
+    $CC -Wall -O2 -o sgrep sgrep.c
+  '';
+
+  installPhase = ''
+    install -D sgrep "$out/bin/sgrep"
+  '';
+
+  installCheckPhase = ''
+    set +o pipefail
+    $out/bin/sgrep 2>&1 | grep ^Usage:
+  '';
+
+  doInstallCheck = true;
+
+  meta = with lib; {
+    homepage = "https://sgrep.sourceforge.net/";
+    description = "Sgrep (sorted grep) searches sorted input files for lines that match a search key";
+    longDescription = ''
+      Sgrep (sorted grep) searches sorted input files for lines that match a search
+      key and outputs the matching lines. When searching large files sgrep is much
+      faster than traditional Unix grep, but with significant restrictions.
+    '';
+    platforms = platforms.unix;
+    license = licenses.gpl3;
```

Nit: I think the [sourceforge page](https://sourceforge.net/projects/sgrep/) says `sgrep` is licensed under GPLv2:
```suggestion
    license = licenses.gpl2;
```

---

I went with GPLv3 because the COPYING file in `sgrep-1.0.tgz` starts with:

```
                    GNU GENERAL PUBLIC LICENSE
                       Version 3, 29 June 2007
```

---

I thought it would be `gpl3` without an explicit "or later version" license application but then I reviewed COPYING and saw

> If the Program does not specify a version number of the GNU General Public License, you may choose any version ever published by the Free Software Foundation.

and in combination with the only "license application" clause found in sgrep.c:

```
 * Copyright 2009 Stephen C. Losen.  Distributed under the terms
 * of the GNU General Public License (GPL)
```

I think you are right and I'll go change it to `gpl3Plus`.